### PR TITLE
refactor(sdk): add tests for whitespace in sweep parameters

### PR DESCRIPTION
**Pull request by kmikowicz-wandb**
Description
-----------
- Fixes WB-13785

In #5523 the user is reporting that the RHS or value of command line arguments should be quoted instead of the whole argument with both the key and value:

eg. agent.name="white space" instead of the current "agent.name=white space"

However, with the new tests that I have written for the agent, I am unable to see any broken behavior with whitespace using the current code. I am continuing to investigate the interaction of the agent arguments and the hydra package.

Testing
-------

Updated test_sweep with arguments to wandb sweep and wandb agent that have whitespace in the parameters. I also manually tested wandb runs which are using hydra and the runs are successful.

### CODE REVIEW

File: tests/system_tests/test_sweep/test_wandb_agent_full.py [hunk 1]
```diff
@@ -60,6 +62,62 @@ def train():
...
+            "method": "grid",
+            "parameters": {"a": {"values": ["one two"]}, "b": {"value": "three four"}},
+        }
+
+        sweep_id = wandb.sweep(sweep_config, project=project)
+        runner.invoke(cli.agent, [sweep_id])
+
+    runs = Api().runs(project, {"sweep": sweep_id})
+    assert len(runs) == 1
+    print("WANDB_RUN_DEBUG:", pprint.pformat(runs[0].metadata))
```
Comment (domphan-wandb) [inline tests/system_tests/test_sweep/test_wandb_agent_full.py]: should this debug be included?

Comment (kmikowicz-wandb) [reply tests/system_tests/test_sweep/test_wandb_agent_full.py]: should remove it, will update
